code/game/objects/items/devices/scanners/health_analyzer.dm

```diff
@@ -342,69 +363,51 @@
-			else
-				render_list += "<span class='info ml-1'>Blood level: [blood_percent]%, [carbontarget.blood_volume] cl, type: [blood_type]</span><br>"
+	var/mob/living/carbon/carbontarget = target
+	var/blood_id = carbontarget.get_blood_id()
+	if(blood_id)
+		var/blood_percent = round((carbontarget.blood_volume / BLOOD_VOLUME_NORMAL) * 100)
+		var/blood_type = carbontarget.dna.blood_type
+		if(blood_id != /datum/reagent/blood) // special blood substance
+			var/datum/reagent/real_reagent = GLOB.chemical_reagents_list[blood_id]
+			blood_type = real_reagent?.name || blood_id
+		if(carbontarget.blood_volume <= BLOOD_VOLUME_SAFE && carbontarget.blood_volume > BLOOD_VOLUME_OKAY)
+			render_list += "<span class='alert ml-1'>Blood level: LOW [blood_percent]%, [carbontarget.blood_volume] cl,</span> [span_info("type: [blood_type]")]<br>"
+		else if(carbontarget.blood_volume <= BLOOD_VOLUME_OKAY)
+			render_list += "<span class='alert ml-1'>Blood level: <b>CRITICAL [blood_percent]%</b>, [carbontarget.blood_volume] cl,</span> [span_info("type: [blood_type]")]<br>"
+		else
+			render_list += "<span class='info ml-1'>Blood level: [blood_percent]%, [carbontarget.blood_volume] cl, type: [blood_type]</span><br>"
 
-	// Blood Alcohol Content
 	var/blood_alcohol_content = target.get_blood_alcohol_content()
 	if(blood_alcohol_content > 0)
 		if(blood_alcohol_content >= 0.24)
 			render_list += "<span class='alert ml-1'>Blood alcohol content: <b>CRITICAL [blood_alcohol_content]%</b></span><br>"
 		else
 			render_list += "<span class='info ml-1'>Blood alcohol content: [blood_alcohol_content]%</span><br>"
 
-	// Cybernetics
-	if(iscarbon(target))
-		var/mob/living/carbon/carbontarget = target
-		var/cyberimp_detect
-		for(var/obj/item/organ/internal/cyberimp/cyberimp in carbontarget.organs)
-			if(IS_ROBOTIC_ORGAN(cyberimp) && !(cyberimp.organ_flags & ORGAN_HIDDEN))
-				cyberimp_detect += "[!cyberimp_detect ? "[cyberimp.examine_title(user)]" : ", [cyberimp.examine_title(user)]"]"
-		if(cyberimp_detect)
-			render_list += "<span class='notice ml-1'>Detected cybernetic modifications:</span><br>"
-			render_list += "<span class='notice ml-2'>[cyberimp_detect]</span><br>"
-	// we handled the last <br> so we don't need handholding
+	//Diseases
+	var/disease_hr = FALSE
+	for(var/datum/disease/disease as anything in target.diseases)
+		if(disease.visibility_flags & HIDDEN_SCANNER)
+			continue
+		if(!disease_hr)
+			render_list += "<hr>"
+			disease_hr = TRUE
+		render_list += "<span class='alert ml-1'>\
+			<b>Warning: [disease.form] detected</b><br>\
+			<div class='ml-2'>\
+			Name: [disease.name].<br>\
+			Type: [disease.spread_text].<br>\
+			Stage: [disease.stage]/[disease.max_stages].<br>\
+			Possible Cure: [disease.cure_text]</div>\
+			</span>"
+
+	// Time of death
+	if(target.station_timestamp_timeofdeath && (target.stat == DEAD || (HAS_TRAIT(target, TRAIT_FAKEDEATH) && !advanced)))
```

Comment:
not really an issue but you sure you want to continue allowing advanced analyzers to detect lings?